src/main/java/net/minecraftforge/common/extensions/IForgeBlockState.java

```diff
@@ -603,18 +605,40 @@ default boolean shouldDisplayFluidOverlay(BlockAndTintGetter level, BlockPos pos
     }
 
     /**
-     * Returns the state that this block should transform into when right clicked by a tool.
-     * For example: Used to determine if an axe can strip, a shovel can path, or a hoe can till.
-     * Return null if vanilla behavior should be disabled.
+     * Returns the state that this block should transform into when right-clicked by a tool.
+     * For example: Used to determine if {@link ToolActions#AXE_STRIP an axe can strip},
+     * {@link ToolActions#SHOVEL_FLATTEN a shovel can path}, or {@link ToolActions#HOE_TILL a hoe can till}.
+     * Returns {@code null} if nothing should happen.
+     *
+     * @param context The use on context that the action was performed in
+     * @param toolAction The action being performed by the tool
+     * @param simulate If {@code true}, no actions that modify the world in any way should be performed. If {@code false}, the world may be modified.
+     * @return The resulting state after the action has been performed
+     */
+    @Nullable
+    default BlockState getToolModifiedState(UseOnContext context, ToolAction toolAction, boolean simulate)
+    {
+        BlockState eventState = net.minecraftforge.event.ForgeEventFactory.onToolUse(self(), context, toolAction, simulate);
+        return eventState != self() ? eventState : self().getBlock().getToolModifiedState(self(), context, toolAction, simulate);
+    }
+
+    /**
+     * Returns the state that this block should transform into when right-clicked by a tool.
+     * For example: Used to determine if {@link ToolActions#AXE_STRIP an axe can strip},
+     * {@link ToolActions#SHOVEL_FLATTEN a shovel can path}, or {@link ToolActions#HOE_TILL a hoe can till}.
```

Comment:
This is inaccurate as it passes through the deprecated method first before trying to do the default hoe till action.

Reply:
On further thought, the blockstate deprecated method won't be called for hoe till, but the block method always will be. Now, do we need to care? The only time this becomes an issue is if people try to coremod into this deprecated IForgeBlockState method, which would require direct JS ASM since Mixins can't target interfaces. So I don't think we need to worry about this javadoc being technically slightly inaccurate.